### PR TITLE
Fix the mobile menu wrongly expanding the logo

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue https://github.com/joomla/joomla-cms/pull/33754#issuecomment-841373252 and #34105

### Summary of Changes

> Joomla logo is expanded pushing the cog icon out of view and toolbar options are displayed by default.

### Testing Instructions

Shrink the width of your browser to a mobile width and open the menu/toolbar. The logo should not expand

### CODE REVIEW

**Comment by Quy** [comment]
It is fine/collapsed on the Home Dashboard initially, but going to other sections such as Articles list, it expands again.

**Comment by RickR2H** [comment]
When I do a refresh in Chrome with the mobile view active, the logo is initially expanded. When I resize the mobile window just a little bit, the logo shrinks and everything jumps into place. When I do a refresh again the logo is back to its initial expanded state.

**Comment by PhilETaylor** [comment]
Is this animation by design? its annoying haha


https://user-images.githubusercontent.com/400092/119253978-a5c6c100-bbab-11eb-8961-df6542f1fe52.mp4

**Comment by PhilETaylor** [comment]
only other thing - which I dont think is related to this PR is

On mobile, navigate to the User Action Log page 
Click bottom right corner to launch menu
Note the menu is open - correct
Note you cannot get to HomeDashboard by scrolling - broken, but probably ok as you can collapse the open Users menu to see it. (Home Dashboard is off screen, and you cannot scroll to is, you have to collapse the open menu leaf to get to it)

**Comment by PhilETaylor** [comment]
> When I do a refresh in Chrome with the mobile view active, the logo is initially expanded. When I resize the mobile window just a little bit, the logo shrinks and everything jumps into place. When I do a refresh again the logo is back to its initial expanded state.

I cannot replicate this today. On a real mobile.

**Comment by PhilETaylor** [comment]
> It is fine/collapsed on the Home Dashboard initially, but going to other sections such as Articles list, it expands again.

I cannot replicate this today.

**Comment by PhilETaylor** [comment]
I have tested this item :white_check_mark: successfully on b0d82e075ffca1ebd11cbfb7f728767c3bcb8259<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33877">issues.joomla.org/tracker/joomla-cms/33877</a>.</sub>

**Comment by PhilETaylor** [comment]
>  When I resize the mobile window 

What I will say, is that the web is meant to be browsed not squished. 

You are correct that there are some issues that a refresh of the page fixes, however its very rare to resize from a desktop view to a mobile view and back to a desktop view without refreshing and so testing that is not really a valid test I dont believe. 


https://user-images.githubusercontent.com/400092/119254487-7e252800-bbae-11eb-946e-d9a763ac4824.mp4

**Comment by dgrammatiko** [comment]
> What I will say, is that the web is meant to be browsed, not squished.

There are still assumptions with the current code, especially for the transition to mobile or going from mobile to desktop. 

Could we tackle those?

Probably yes by transforming the cookie value to a JSON string (store state for logo, menu state and header toolbar), but I see little advantage to complicate further the code to iron out a couple of glitches which are edge cases (e.g. clicking on the save button on mobile view will not reload the page with the toolbar open). There are trade-offs (perf vs covering all cases), and as is the script is not very efficient/performant, and my whole approach here was to eliminate as much as possible the annoying open/close animations on page reload and not to re-architecture the way we put together those underlaying parts

**Comment by PhilETaylor** [comment]
> however its very rare to resize from a desktop view to a mobile view and back to a desktop view without refreshing and so testing that is not really a valid test I dont believe.

^^ thats my way of saying I agree with you, its a non-issue at this stage. If you are going to rotate a mobile device from landscape to portrait and back again I dont expect a website to handle that perfectly, even in 2021, and its no harm to refresh.

File: build/media_source/templates/administrator/atum/js/template.es6.js [hunk 1]
```diff
@@ -149,11 +154,11 @@ function setMobile() {
   }
 
   if (small.matches) {
```
Comment (eopws) [inline build/media_source/templates/administrator/atum/js/template.es6.js:156]: I think we should have `mobile.matches` here as it wouldn't work on tablets and mobiles

Comment (dgrammatiko) [reply build/media_source/templates/administrator/atum/js/template.es6.js:156]: Tablets are fine with the collabsibles open, they have enough space...

File: build/media_source/templates/administrator/atum/js/template.es6.js [hunk 2]
```diff
@@ -217,11 +222,23 @@ function subheadScrolling() {
 headerItemsInDropdown();
 reactToResize();
 subheadScrolling();
+if (mobile.matches) {
```
Comment (eopws) [inline build/media_source/templates/administrator/atum/js/template.es6.js:225]: Why not using `setMobile` function call inside this block?

File: build/media_source/templates/administrator/atum/js/template.es6.js (hunk 1 above)
Comment (eopws) [reply build/media_source/templates/administrator/atum/js/template.es6.js:156]: What about phones in a horizontal position?

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on cdfbb5f951ff6c2b7017905ddcd2294da96ed814<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33877">issues.joomla.org/tracker/joomla-cms/33877</a>.</sub>

**Comment by PhilETaylor** [comment]
I have tested this item :white_check_mark: successfully on cdfbb5f951ff6c2b7017905ddcd2294da96ed814<br /><br />A quick test as I tested intensively previously <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33877">issues.joomla.org/tracker/joomla-cms/33877</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33877">issues.joomla.org/tracker/joomla-cms/33877</a>.</sub>

**Comment by chmst** [comment]
Thanks!

**Pull request merged**